src/gc/collector.cpp

```diff
@@ -116,16 +139,86 @@ class TraceStack {
 
 
     void* pop() {
-        if (cur > start)
-            return *--cur;
-
-        return pop_chunk_and_item();
+        if (cur > start) {
+            previous_pop = *--cur;
+        } else {
+            previous_pop = pop_chunk_and_item();
+        }
+        return previous_pop;
     }
 };
 std::vector<void**> TraceStack::free_chunks;
 
+class MarkStack : public TraceStack {
+public:
+    MarkStack(const std::unordered_set<void*>& root_handles) : TraceStack() {
+        for (void* p : root_handles) {
+            assert(!isMarked(GCAllocation::fromUserData(p)));
+            push(p);
+        }
+    }
+
+    virtual bool visit_action(GCAllocation* al) {
+// Use this to print the directed edges of the GC graph traversal.
+// i.e. print every a > b where a is a pointer and b is something a references
+#if 0
```

Comment:
could this be enabled, but in a separate subclass/instantiation of MarkStack?  I'd very much like `pyston::gc::whatReferences(void* p)` that I can use in gdb.